File: src/Unidesk/Unidesk/Dtos/Requests/ThesisFilter.cs
using Unidesk.Db.Models;

namespace Unidesk.Dtos.Requests;

public class ThesisFilter : IFilter
{
    public QueryPaging Paging { get; set; }
    public Guid? UserId { get; set; }
    public List<Guid> Keywords { get; set; } = new();
    public Operator Operator { get; set; }
    
    public ThesisStatus? Status { get; set; }
    
    public bool? HasKeywords { get; set; }
    public string? Keyword { get; set; }
    
    public Guid? SchoolYearId { get; set; }
    
    public bool MyThesis { get; set; }
}

File: src/Unidesk/Unidesk.Client/src/api-client/models/ThesisFilter.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { QueryFilter } from './QueryFilter';
import type { ThesisStatus } from './ThesisStatus';

export type ThesisFilter = {
    filter?: QueryFilter;
    userId?: string | null;
    keywords?: Array<string> | null;
    status?: ThesisStatus;
    hasKeywords?: boolean | null;
    keyword?: string | null;
};
